src/dpki.rs

```diff
@@ -0,0 +1,377 @@
+
+// DPKI-compatible pure-rust intrinsics for dealing in Seeds, Keypairs, password hashes
+// 
+// We'll use pure-rust cryptographic primatives, to ease WASM generation.  Later, we can transition
+// to using Holochain's DPKI API.
+
+
+use std::fmt;
+use failure::Error;
+use hcid;
+use bip39::{
+    Language, Mnemonic, MnemonicType
+};
+use sha2::{
+    Sha512, Digest
+};
+use serde::{
+    Deserialize, Deserializer, Serialize, Serializer, de
+};
+
+use ed25519_dalek::{
+    PublicKey, SecretKey, Keypair, SignatureError,
+};
+
+pub const PUBLIC_KEY_SIZE: usize = 256/8;
+pub const SECRET_KEY_SIZE: usize = 256/8;
+
+pub const AEAD_TAG_SIZE: usize = 256/8;
+
+pub const SEED_SIZE: usize = 256/8;
+pub const ENCRYPTED_SEED_SIZE: usize = SEED_SIZE + AEAD_TAG_SIZE;
+
+#[derive(Debug)]
+pub struct SigningKeyPair(pub Keypair);
+#[derive(Debug)]
+pub struct SigningPublicKey(pub PublicKey);
+#[derive(Debug)]
+pub struct SigningSecretKey(pub SecretKey);
```

Comment:
Same here, I don't think we should wrap any of that into custom types.

Reply:
We cannot have custom serialization unless we create a type wrapper.